File: Shubhankar/Customer/Server/src/models/customer.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToMany } from "typeorm";
import { CustomerGroup } from "./customergroup";
@Entity()
export class Customer {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column()
  name!: string;

  @Column()
  status!: string;

  @Column()
  address!: string;

  @Column({ type: "varchar", length: 10 })
  phonenumber!: number;

  @ManyToMany(() => CustomerGroup, (customergroup) => customergroup.customers)
  customergroups!: Customer[];
}

File: harsha/Customer/server/src/main/java/com/example/onetoonemapping/models/Customer.java
package com.example.onetoonemapping.models;

import java.util.List;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "customers")
public class Customer {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	private String name;
	private String status;
	private String address;
	private long phoneNo;

	@JsonBackReference
	@ManyToMany(mappedBy = "customers")
	private List<CustomerGroup> customerGroups;
}